### ISSUE TITLE
IntrospectionProcessor __invoke Doesn't Hit With the Stack Logging Channel

### ISSUE THREAD

**Issue by chunterb** [danielme85/laravel-log-to-db#5]
Hello,

First off, thank you for this awesome package - it is saving me a lot of work. I am having to customize it to our needs, but it works really well to begin with.

I have stumbled across a  problem, and this is tested on a fresh install to assure that it is not my mistake. 

The problem is that when you turn on the 'stack' channel for logging, the IntrospectionProcessor ``` __invoke``` never gets hit, causing all of the logs to exclude the IntrospectionProcessor (file, class, etc..).

It seems that it tries to hit the ``` local ``` channels, before the ``` database ``` channel. After that, it uses the formatting for the local channel instead of the formatting for the DB logger. Not sure if this could be overridden - if that's the case.

Am I missing something here?

**Comment by danielme85** [comment]
Hi @chunterb, 
I think you are on to something here, it does seem to behave differently when using the 'stack' channel.

```php
/**
 * Class LogToDbHandler
 *
 * @package danielme85\LaravelLogToDB
 */
class LogToDbHandler
{
    /**
     * Create a custom Monolog instance.
     *
     * @param  array  $config
     * @return \Monolog\Logger
     */
    public function __invoke(array $config)
    {
        return new Logger(
            $config['name'] ?? 'LogToDB', // <- this is missing as well :(
            [
                // The log handler class gets passed along fine
                new LogToDbCustomLoggingHandler(
                    $config['connection'] ?? null,
                    $config['collection'] ?? null,
                    $config['detailed'] ?? null,
                    $config['queue'] ?? null,
                    $config['queue_name'] ?? null,
                    $config['queue_connection'] ?? null,
                    $config['level'] ?? null)
            ],
            [
                new IntrospectionProcessor() // <- there is our missing friend :(
            ]);
    }
}
```
The $config['name'] is missing when new Logger class is called trough the 'stack' channel, so the same thing might be happening to the IntrospectionProcessor...? 
I'l try to look into this sometime this week: my first guess would be that maybe there is some 'stack' logic that calls a new Logger class after LogToDbHandler and/or maybe the Logger class is passed along from channel to channel?

Thank you for letting me know about this issue, and for your compliment (you might want to save that for when we actually get this working correctly 😛).
Any help/feedback would definitely be welcome!

**Comment by chunterb** [comment]
@danielme85 

So, I have found this much...

When using the stack driver with the database channel, it looks to be calling a different handler.

To see for yourself:

``` Line 208 on Illuiminate\Log\LogManager.php ```
``` php 
    /**
     * Create an aggregate log driver instance.
     *
     * @param  array  $config
     * @return \Psr\Log\LoggerInterface
     */
    protected function createStackDriver(array $config)
    {
        $handlers = collect($config['channels'])->flatMap(function ($channel) {
            
            if ($channel == 'database')
            {
                dd($this->channel($channel));
            }

            return $this->channel($channel)->getHandlers();
        })->all();

        if ($config['ignore_exceptions'] ?? false) {
            $handlers = [new WhatFailureGroupHandler($handlers)];
        }

        return new Monolog($this->parseChannel($config), $handlers);
    }
```

This outputs:

``` php
Logger {#199 ▼
  #logger: Logger {#194 ▼
    #name: "DB Logger"
    #handlers: array:1 [▼
      0 => LogToDbCustomLoggingHandler {#197 ▶} <----- Notice this
    ]
    #processors: array:1 [▶]
    #microsecondTimestamps: true
    #exceptionHandler: null
  }
  #dispatcher: Dispatcher {#28 ▶}
}
```

The handler that is normally called is ``` LogToDbHandler.php ``` , but here you see it's using the other one.

Now I'm trying to find where exactly it gets the handlers, which you may already know.

Hope this is somewhat helpful.

**Comment by danielme85** [comment]
Thanks, that seems like the perfect place to start my debugging.
Please let me know if you have any success, and I will do the same.

**Comment by danielme85** [comment]
Could you checkout the dev branch and see if that works better for you? c725ae513413557f53725ffc4a554f27ccd204ed

Turns out there is a pushProcessor() function on the Monolog Handler class, hopefully that would fix it.

**Comment by danielme85** [comment]
Hopefully fixed by 14bb8fc49d13a89036e3d42d1ed179c51fd70837
